src/acceptor.rs

```diff
@@ -104,8 +158,16 @@ impl Acceptor {
 }
 
 impl Drop for Acceptor {
+    #[cfg(not(test))]
+    fn drop(&mut self) {
+        self.running.store(false, Ordering::SeqCst);
```

Comment:
This is better handled using the net::Shutdown Paradigm. You can use this example https://github.com/maidsafe/crust/blob/new_refactor/src/connection.rs#L113